components/backends/trtllm/src/dynamo/trtllm/multimodal_processor.py

```diff
@@ -46,6 +54,35 @@ def __init__(
         self.tokenizer = tokenizer
         self.modality = ""
 
+    def is_url(self, path: str) -> bool:
+        """Check if a path is a URL."""
+        parsed = urlparse(path)
+        return bool(parsed.scheme and parsed.netloc)
+
+    def load_tensor_from_path_or_url(self, path: str) -> torch.Tensor:
+        """Load a tensor from either a local file path or a URL."""
+        if self.is_url(path):
+            # Download the file to a temporary location
+            with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+                try:
+                    urlretrieve(path, tmp_file.name)
+                    tensor = torch.load(tmp_file.name, map_location="cpu")
+                    return tensor
```

Comment:
torch.load should support byte streams / ByteIO, you can avoid triggering file system ops  entirely